engine/runtime-parser-dsl/src/main/java/org/enso/runtime/parser/dsl/GenerateIR.java

```diff
@@ -0,0 +1,31 @@
+package org.enso.runtime.parser.dsl;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * A class annotated with this annotation will be processed by the IR processor. The processor will
+ * generate a super class that is meant to be extended by this class. The generated class will have
+ * the same package as this class, and its name will have the "Gen" suffix. Majority of the methods
+ * in the generated class will be either private or package-private, so that they are not accessible
+ * from the outside.
+ *
+ * <p>The class can be enclosed (nested inside) an interface.
+ *
+ * <p>The class must contain a single constructor annotated with {@link GenerateFields}.
+ */
+@Retention(RetentionPolicy.SOURCE)
+@Target(ElementType.TYPE)
+public @interface GenerateIR {
+
+  /**
+   * Name of the interfaces that the generated superclass must implement. All the interfaces must be
+   * subtypes of the {@code org.enso.compiler.ir.IR} interface. All the abstract parameterless
+   * methods from all the interfaces will be implemented by the generated class. Must not be empty.
+   *
+   * @return
+   */
+  String interfaces() default "org.enso.compiler.core.IR";
```

Comment:
I guess this is not used yet. I see plural being used here. In such case the value should be `[]` array. I would also suggest to use `Class[]` that way one prevents typos as the referenced classes must exist. Btw. looking at `Block` I see it implements `IRKind.Primitive` - that interface isn't subtype of `IR`...

Reply:
`GenerateIR.interfaces` refactored to `Class[]` in e54fdb933026929d618c3855689657ef57059d98.

Reply:
In 9da9394508cb519b6035d559d4230e773ef9bab1, one can specify multiple `interfaces`. Moreover, I have updated the docs and tests. There is no need for the `interfaces` to be subtypes of `org.enso.compiler.core.IR` anymore.